src/Pim/Component/ReferenceData/Factory/ProductValue/ReferenceDataCollectionProductValueFactory.php

```diff
@@ -116,21 +112,42 @@ protected function checkData(AttributeInterface $attribute, $data)
     }
 
     /**
+     * Gets a collection of reference data from an array of codes.
+     *
+     * @param AttributeInterface $attribute
+     * @param array              $referenceDataCodes
+     *
+     * @return ArrayCollection
+     */
+    protected function getReferenceDataCollection(AttributeInterface $attribute, array $referenceDataCodes)
+    {
+        $collection = new ArrayCollection();
+
+        $repository = $this->repositoryResolver->resolve($attribute->getReferenceDataName());
+
+        foreach ($referenceDataCodes as $referenceDataCode) {
+            $collection->add($this->getReferenceData($attribute, $repository, $referenceDataCode));
+        }
+
+        return $collection;
+    }
+
+    /**
      * Finds a reference data by code.
      *
-     * @param AttributeInterface               $attribute
-     * @param ReferenceDataRepositoryInterface $repository
-     * @param string                           $referenceDataCode
+     * @param AttributeInterface                    $attribute
+     * @param IdentifiableObjectRepositoryInterface $repository
```

Comment:
if I understand well, that can't work
that would mean the `ReferenceDataRepositoryInterface` implements `IdentifiableObjectRepositoryInterface` 
and I'm not sure we want this constraint for the integrators